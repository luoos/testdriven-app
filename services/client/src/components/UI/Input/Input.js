import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => {

  let size;
  switch (props.size) {
    case 'small':
      size = 'is-small';
      break;
    case 'medium':
      size = 'is-medium';
      break
    case 'large':
      size = 'is-large';
      break
    default:
      size = 'is-medium';
  }

  return (
    <div className="field">
      <label className={`label ${size}`}>{props.label}</label>
      <div className={'control ' + (props.icon && 'has-icons-left')}>
        <input
          className={`input ${size}`}
          name={props.name}
          type={props.text}
          placeholder={props.placeholder}
          required={props.required}
          value={props.value}
          onChange={props.handleChange}
        />
        {
          props.icon &&
          <span className="icon is-left">
            <i className={`fas fa-${props.icon}`}></i>
          </span>
        }
      </div>
    </div>
  );
}

Input.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,

  size: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
}

export default Input;