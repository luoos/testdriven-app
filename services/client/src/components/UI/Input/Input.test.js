import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Input from './Input';

let testProps;
let componenet;

describe('<Input />', () => {

  beforeEach(() => {
    testProps = {
      label: 'test-label',
      name: 'test-name',
      type: 'text',
      icon: 'test-icon',
      value: 'test-value',
      handleChange: jest.fn(),
      size: 'large',
      placeholder: 'enter some',
      required: true
    }
    componenet = <Input {...testProps} />
  })

  it('Input calls handleChange properly', () => {
    const wrapper = shallow(componenet);
    expect(testProps.handleChange).toHaveBeenCalledTimes(0);
    const input = wrapper.find('input');
    expect(input.length).toBe(1);
    input.simulate('change',
      {target: {name: 'test-name', value: 'hello there'} });
    expect(testProps.handleChange).toHaveBeenCalledTimes(1);
  });

  it('renders a snapshot properly', () => {
    const tree = renderer.create(componenet).toJSON();
    expect(tree).toMatchSnapshot();
  })

});
