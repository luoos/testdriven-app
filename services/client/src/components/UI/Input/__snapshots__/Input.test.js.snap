// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders a snapshot properly 1`] = `
<div
  className="field"
>
  <label
    className="label is-large"
  >
    test-label
  </label>
  <div
    className="control has-icons-left"
  >
    <input
      className="input is-large"
      name="test-name"
      onChange={[MockFunction]}
      placeholder="enter some"
      required={true}
      value="test-value"
    />
    <span
      className="icon is-left"
    >
      <i
        className="fas fa-test-icon"
      />
    </span>
  </div>
</div>
`;
